test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
hennna@ Are you planning on making the upstream IPv6 as well in this PR or pushing it to later? I think it can wait, since getting just basic client <-> Envoy IPv6 happening is a self-contained effort. But, we should probably add some TODOs in the code base to keep track that we also need to validate the upstream in the integration tests.

There's also the question of whether we want to validate IPv4 downstream and IPv6 upstream combos for example, or all IPv4 and all IPv6. I think the latter provides sufficient coverage and will simplify the TEST_P, but worth discussing.